.github/workflows/perf.yml

```diff
@@ -0,0 +1,122 @@
+      aws-bucket:
+        type: string
+        required: true
+        description: The AWS bucket as output by 'output.bucket_name'
+      ref:
+        type: string
+        required: false
+        description: The ref of the test-plans repo to use (defaults to 'master')
+        default: master
+    secrets:
+      PERF_AWS_SECRET_ACCESS_KEY: # The AWS secret access key of 'aws_iam_user.perf'
+        required: true
+      PERF_SSH_PRIVATE_KEY: # The SSH private key for 'aws_key_pair.perf'
+        required: true
+
+jobs:
+  perf:
+    name: Perf
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+    defaults:
+      run:
+        shell: bash
+        working-directory: perf
+    env:
+      AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id || vars.PERF_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.PERF_AWS_SECRET_ACCESS_KEY }}
+    steps:
+      - name: Configure SSH
+        uses: webfactory/ssh-agent@d4b9b8ff72958532804b70bbe600ad43b36d5f2e # v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.PERF_SSH_PRIVATE_KEY }}
+      - name: Checkout test-plans
+        uses: actions/checkout@v3
+        with:
+          repository: libp2p/test-plans
+          ref: ${{ inputs.ref || github.ref }}
+      - id: server
+        name: Provision server
+        run: echo "id=$(make provision-server | tail -n 1)" >> $GITHUB_OUTPUT
+      - id: client
+        name: Provision client
+        run: echo "id=$(make provision-client | tail -n 1)" >> $GITHUB_OUTPUT
+      - id: ip
+        name: Wait for client/server IP
+        env:
+          SERVER_ID: ${{ steps.server.outputs.id }}
+          CLIENT_ID: ${{ steps.client.outputs.id }}
+        run: |
+          read SERVER_IP CLIENT_IP <<< $(make wait SERVER_ID=$SERVER_ID CLIENT_ID=$CLIENT_ID | tail -n 1)
+          echo "server=$SERVER_IP" >> $GITHUB_OUTPUT
+          echo "client=$CLIENT_IP" >> $GITHUB_OUTPUT
+      - name: Download dependencies
+        run: npm ci
+        working-directory: perf/runner
+      - name: Run tests
+        env:
+          SERVER_IP: ${{ steps.ip.outputs.server }}
+          CLIENT_IP: ${{ steps.ip.outputs.client }}
+        run: npm run start -- --client-public-ip $CLIENT_IP --server-public-ip $SERVER_IP
+        working-directory: perf/runner
+      - name: Archive results
+        uses: actions/upload-artifact@v2
+        with:
+          name: results
+          path: perf/runner/benchmark-results.json
+      - id: s3
+        name: Upload results
+        env:
+          AWS_BUCKET: ${{ inputs.aws-bucket || vars.PERF_AWS_BUCKET }}
+          AWS_BUCKET_PATH: ${{ github.repository }}/${{ github.run_id }}/${{ github.run_attempt }}/benchmark-results.json
+        run: |
+          aws s3 cp benchmark-results.json s3://$AWS_BUCKET/$AWS_BUCKET_PATH --acl public-read --region us-west-2
+          echo "url=https://$AWS_BUCKET.s3.amazonaws.com/$AWS_BUCKET_PATH" >> $GITHUB_OUTPUT
+        working-directory: perf/runner
```

Comment:
I don't think these steps are necessary. See comment above.

Reply:
Sure! I'll replace it with a push.